packages/cadl-python/src/emitter.ts

```diff
@@ -247,15 +242,16 @@ function getEffectiveSchemaType(program: Program, type: Model): Model {
 function getType(program: Program, type: EmitterType): any {
     // don't cache simple type(string, int, etc) since decorators may change the result
     const enableCache = !isSimpleType(program, type);
+    const effectiveModel = type.kind === "Model" ? getEffectiveSchemaType(program, type) : type;
```

Comment:
we are using effective type in `emitModel` (https://github.com/Azure/autorest.python/blob/fbcadb4c739d4b7c8edab21a0fa66791b1bd40c2/packages/cadl-python/src/emitter.ts#L722), so we need changes here for caching by effective type,